projects/power_distribution/src/smoke_uv_cutoff.c

```diff
@@ -0,0 +1,46 @@
+#include "smoke_uv_cutoff.h"
+
+#include "can_transmit.h"
+#include "front_uv_detector.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "pd_events.h"
+#include "pin_defs.h"
+#include "status.h"
+#include "wait.h"
+
+static void prv_pin_interrupt_handler(const GpioAddress *address, void *context) {
+  CAN_TRANSMIT_UV_CUTOFF_NOTIFICATION();
+
+  LOG_WARN("UV lockout was triggered\n");
+}
+
+StatusCode front_uv_detector_init(GpioAddress *detector_pin) {
+  InterruptSettings interrupt_settings = {
+    .type = INTERRUPT_TYPE_INTERRUPT,
+    .priority = INTERRUPT_PRIORITY_NORMAL,
+  };
+
+  // callback occurs when pin turns low i.e falling
+  status_ok_or_return(gpio_it_register_interrupt(
+      detector_pin, &interrupt_settings, INTERRUPT_EDGE_FALLING, prv_pin_interrupt_handler, NULL));
+
+  return STATUS_CODE_OK;
+}
```

Comment:
It looks like you've copy-pasted front_uv_cutoff.c here - you don't need to do that, you just need to call `front_uv_detector_init` in `smoke_uv_cutoff_perform` to initialize the module, and it'll do the rest.

Use `FRONT_UV_COMPARATOR_PIN` from pin_defs.h as the GpioAddress parameter - you might have to assign it to a variable and pass in its address though, or use the syntax `&(GpioAddress)FRONT_UV_COMPARATOR_PIN`, which works due to another GNU extension called "compound literals" (google it).

When you call `front_uv_detector_init`, you'll also want to wrap its call in `BUG()` - it's a macro from bug.h that will cause a big fancy error message to be printed if the wrapped call returns a StatusCode which isn't `STATUS_CODE_OK` (any non-OK value means that the initialization failed). Since this is a smoke test, we want lots of logging and to make it easy to figure out if something went wrong in software.